src/System.Net.Http/src/netcore50/System/Net/HttpClientHandler.cs

```diff
@@ -226,6 +228,45 @@ public ICredentials Credentials
             }
         }
 
+        public ICredentials DefaultProxyCredentials
+        {
+            get
+            {
+                RTPasswordCredential rtCreds = rtFilter.ProxyCredential;
+                if (rtCreds == null)
+                {
+                    return null;
+                }
+
+                NetworkCredential creds = new NetworkCredential(rtCreds.UserName, rtCreds.Password);
+                return creds;
+            }
+            set
+            {
+                if (value == null)
+                {
+                    CheckDisposedOrStarted();
```

Comment:
I assume you're doing this in each block rather than at the beginning once because you want the PNSE to take precedence?